File: go/Poor_Pigs.go
package main

import (
    "fmt"
    "math"
)

func log(x, y int) float64 {
    return math.Log(float64(y)) / math.Log(float64(x))
}

func poorPigs(buckets int, minutesToDie int, minutesToTest int) int {
    return int(math.Ceil(log(1 + minutesToTest / minutesToDie, buckets)))
}

func main() {
    buckets := 1000
    minutesToDie := 15
    minutesToTest := 60
    fmt.Println(poorPigs(buckets, minutesToDie, minutesToTest))
}

File: js/Poor_Pigs.js
/**
 * @param {number} buckets
 * @param {number} minutesToDie
 * @param {number} minutesToTest
 * @return {number}
 */

function log(x, y) {
    return Math.log(y) / Math.log(x);
}
var poorPigs = function(buckets, minutesToDie, minutesToTest) {
    return Math.ceil(log(1 + minutesToTest/minutesToDie, buckets));
};

var buckets = 1000;
var minutesToDie = 15;
var minutesToTest = 60;

console.log(poorPigs(buckets, minutesToDie, minutesToTest));
